xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/pdf/impl/PdfExportImpl.java

```diff
@@ -329,26 +329,14 @@ String applyCSS(String html, String css, XWikiContext context)
             XHTMLDocumentFactory docFactory = XHTMLDocumentFactory.getInstance();
             SAXReader reader = new SAXReader(docFactory);
 
-            // Dom4J 2.1.1 disables external DTDs by default, which is required here, putting it back.
+            // Dom4J 2.1.1 disables external DTDs by default, so we set our own XMLReader.
             // See https://github.com/dom4j/dom4j/issues/51
+            XMLReader xmlReader = Utils.getComponent(XMLReaderFactory.class).createXMLReader();
             try {
-                reader.setFeature("http://apache.org/xml/features/nonvalidating/load-external-dtd", true);
-            } catch (SAXNotSupportedException | SAXNotRecognizedException e) {
-                // We ingore these 2 exceptions to properly handle the case when a non-Xerces parser is provided by
-                // JAXP. Here are the possible flows:
-                //
-                // A) JAXP gives a Xerces instance:
-                // dom4j 2.1.1 sets "http://apache.org/xml/features/nonvalidating/load-external-dtd" to false,
-                // to disable DTD. No exception is thrown.
-                // XWiki sets "http://apache.org/xml/features/nonvalidating/load-external-dtd" to true, to enable DTD
-                // loading. No exception.
-                //
-                // B) JAXP gives a non-Xerces instance:
-                // dom4j 2.1.1 sets "http://apache.org/xml/features/nonvalidating/load-external-dtd" to false. The
-                // feature is not recognized: exception is thrown and ignored.
-                // XWiki sets "http://apache.org/xml/features/nonvalidating/load-external-dtd" to true. Exception is
-                // thrown (and was not being ignored). Boom.
+                xmlReader.setFeature("http://xml.org/sax/features/namespaces", true);
```

Comment:
Could you explain why we set this here?

Reply:
> It sets the http://xml.org/sax/features/namespaces feature which is generally a good idea (and compatible with what dom4j did to its default reader).

Why is it a good idea?

Reply:
Supporting namespaces is a good idea, if you do not know in advance whether the processed XML is going to use namespaces. In any case, as mentioned in the commit message this patch is replacing something that **did** support namespaces. Removing namespace support is what should be justified, but I'm just keeping what was being done.

I did not explain it in comments because the older namespace default wasn't either (it was a dom4j default).

Reply:
> I did not explain it in comments because the older namespace default wasn't either (it was a dom4j default).

Sure but that's a big difference! Before, we didn't have to care about it and we were trusting the 3rd party software to do the right thing, to have its own tests and its own documentation.

With your change it now becomes XWiki's responsibility and the extra code is now on xwiki's side, so we need to do the work of explaining why we do that (and ideally even test it to verify it works, which is something we didn't need to do before).

So I suggest we add your comment to the code "Supporting namespaces is a good idea, if you do not know in advance whether the processed XML is going to use namespaces.".

Last, I think this should be done in a central place so that all code requiring an XMLReader shouldn't have to redo it. WDYT? Don't we have that in commons-xml already?

Thanks